network/p2p/middleware/middleware.go

```diff
@@ -512,6 +518,15 @@ func (m *Middleware) handleIncomingStream(s libp2pnetwork.Stream) {
 			return
 		}
 
+		// ignore messages if node does not have subscription to topic
+		channel := channels.Channel(msg.ChannelID)
+		topic := channels.TopicFromChannel(channel, m.rootBlockID)
+		if !m.libP2PNode.HasSubscription(topic) {
+			violation := &slashing.Violation{Identity: nil, PeerID: remotePeer.String(), MsgType: "", Channel: channel, IsUnicast: true, Err: ErrUnicastMsgWithoutSub}
```

Comment:
Can we add the message type as well? (decoding first byte)

Reply:
https://github.com/onflow/flow-go/pull/3490/commits/535e6ae1dec1cabaf3049d69fc0828cb62f187e7